builtin/game/item_entity.lua

```diff
@@ -37,32 +37,35 @@ core.register_entity(":__builtin:item", {
 	slippery_state = false,
 	age = 0,
 
-	set_item = function(self, itemstring)
-		local stack = ItemStack(itemstring or self.itemstring)
+	set_item = function(self, item)
+		local stack = ItemStack(item or self.itemstring)
 		self.itemstring = stack:to_string()
+		if self.itemstring == "" then
+			-- item not yet known
+			return
+		end
 
 		-- Backwards compatibility: old clients use the texture
 		-- to get the type of the item
-		local itemname = stack:get_name()
+		local itemname = stack:is_known() and stack:get_name() or "unknown"
 
 		local max_count = stack:get_stack_max()
 		local count = math.min(stack:get_count(), max_count)
-		local size = 0.2 + 0.1 * (count / max_count)
-
-		if not stack:is_known() then
-			itemname = "unknown"
-		end
+		local size = 0.2 + 0.1 * (count / max_count)^(1/3)
```

Comment:
For accurate volume representng number, this would be something like:
`local size = 0.3 * (count / max_count)^(1/3)`

Reply:
Good idea, paramat.
![grafik](https://user-images.githubusercontent.com/1497498/30005086-faf04106-90da-11e7-8a30-6a8111ee1b7d.png)

Reply:
Oops, might need scaling.

Reply:
Hmm but then size of 1 will differ based on max stack number.

Reply:
Spaces around '^' and '/'

Reply:
The 0.2 represents the overhead, the size of the container which keeps the item as an object.

Reply:
That's ok for me.